exporter/opentelemetry-exporter-datadog/src/opentelemetry/exporter/datadog/exporter.py

```diff
@@ -312,19 +313,23 @@ def _parse_tags_str(tags_str):
     return parsed_tags
 
 
-def _extract_tags_from_resource(resource):
+def _extract_tags_from_resource(resource, fallback_service_name):
     """Parse tags from resource.attributes, except service.name which
     has special significance within datadog"""
     tags = {}
     service_name = None
     if not (resource and getattr(resource, "attributes", None)):
-        return [tags, service_name]
+        return [tags, fallback_service_name]
 
     for attribute_key, attribute_value in resource.attributes.items():
         if attribute_key == SERVICE_NAME_TAG:
             service_name = attribute_value
         else:
             tags[attribute_key] = attribute_value
+
+    if service_name is None or service_name == UNKNOWN_SERVICE_NAME:
```

Comment:
It look like `UNKNOWN_SERVICE_NAME` is only used 1 time in this exporter? Because that's the case you could consider just doing `"unknown_service"` like we do in `opentelemetry-sdk`:

https://github.com/open-telemetry/opentelemetry-python/blob/f11ed2f3bacb11d53a7a2b4837cf6308fa34cc71/opentelemetry-sdk/src/opentelemetry/sdk/resources/__init__.py#L172

But not a blocker!

Reply:
Either that or change both to use `UNKNOWN_SERVICE_NAME`

Reply:
thanks, updated, i just removed the use of `UNKNOWN_SERVICE_NAME` entirely and went with "unknown_service".